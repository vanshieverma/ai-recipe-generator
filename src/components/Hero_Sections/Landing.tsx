import { signIn } from 'next-auth/react';
export default function Landing() {
    return (
        <div className="text-center">
            <h1 className="text-4xl font-bold tracking-tight text-gray-900 sm:text-6xl">
                Generate Delicious Recipes with Your Ingredients
            </h1>
            <p className="mt-6 text-lg leading-8 text-gray-600">
                Simply input your available ingredients, select dietary preferences, and let our AI create unique and delicious recipes just for you.
            </p>
            <div className="mt-10 flex items-center justify-center gap-x-6">
                <button
                    className="rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    onClick={() => signIn('google')}
                >
                    Get started
                </button>
            </div>
        </div>
    );
}