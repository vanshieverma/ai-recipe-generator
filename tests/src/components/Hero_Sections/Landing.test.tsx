import { signIn } from "next-auth/react";
import Landing from "../../../../src/components/Hero_Sections/Landing";
import { render, screen, fireEvent } from '@testing-library/react'


jest.mock("next-auth/react")

it('The Landing section renders', () => {
    const { container } = render(<Landing />)
    expect(container).toMatchSnapshot()
})

it('Shall be the user to sign in', () => {
    const mockSignIn = jest.fn();
    (signIn as jest.Mock).mockImplementationOnce(mockSignIn);
    const { container } = render(<Landing />)
    fireEvent.click(screen.getByText('Get started'));
    expect(mockSignIn).toHaveBeenCalledTimes(1)
})