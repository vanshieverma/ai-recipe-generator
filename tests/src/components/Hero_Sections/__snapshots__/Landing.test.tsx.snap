// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Landing section renders 1`] = `
<div>
  <div
    class="text-center"
  >
    <h1
      class="text-4xl font-bold tracking-tight text-gray-900 sm:text-6xl"
    >
      Generate Delicious Recipes with Your Ingredients
    </h1>
    <p
      class="mt-6 text-lg leading-8 text-gray-600"
    >
      Simply input your available ingredients, select dietary preferences, and let our AI create unique and delicious recipes just for you.
    </p>
    <div
      class="mt-10 flex items-center justify-center gap-x-6"
    >
      <button
        class="rounded-md bg-indigo-600 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
      >
        Get started
      </button>
    </div>
  </div>
</div>
`;
